### ISSUE TITLE
Support nullable String preferences, i.e. use null as default value

### ISSUE THREAD

**Issue by cypressious** [nsk-mironov/kotlin-jetpack#15]
Like the title says

**Comment by nsk-mironov** [comment]
It's already implemented in master, will be available in next release.

**Comment by nsk-mironov** [comment]
Btw, what do think about having nullable preferences with a not null default value? (meaning it will have some value by default, but you still can explicitly set the value to `null`)

``` kotlin
public class Example(private val preferences: SharedPreferences) {
    private val value by preferences.bindOptionalPreference<String>(default = "not null")

    public fun example() {
        preferences.editor().clear().apply()

        // prints "value = not null"
        Log.d("Example", "value = ${value}")

        value = null

        // prints "value = "
        Log.d("Example", "value = ${value.orEmpty()}")
    }
}
```

This is not supported right now and I'm not sure if somebody ever needs this.

**Comment by cypressious** [comment]
I think it's not necessary.

**Comment by nsk-mironov** [comment]
Fixed in `0.13.0`
